### DISCUSSION TITLE
Blocky as external proxy for browser

### DISCUSSION THREAD

**Discussion by PowershellScripter** [0xERR0R/blocky#301] [Q&A]
Am I able to setup blocky behind a reverse proxy and use the url to it as a proxy inside my browser to block ads with?

**Comment by 0xERR0R** [comment]
Hi,

Yes, you need a reverse proxy with properly configured SSL certificate (e.g. Let's Encrypt). Than you can use blocky as DoH (DNS-over-HTTPS) resolver in your browser. Here is documentation for Firefox: https://support.mozilla.org/en-US/kb/firefox-dns-over-https

**Reply by PowershellScripter** [reply]
Awesome! Thank you for the info. I have traefik setup in my eco system now since I have somewhat more advanced needs, so traefik is setup to handle all my SSL needs. Im just looking for something to be able to use as an adblock proxy so I dont have to use a vpn to connect to the network I have my adblocker hosted on but still be able to apply adblocking when needed.

So, thank you for the info!



> Hi,
> 
> Yes, you need a reverse proxy with properly configured SSL certificate (e.g. Let's Encrypt). Than you can use blocky as DoH (DNS-over-HTTPS) resolver in your browser. Here is documentation for Firefox: https://support.mozilla.org/en-US/kb/firefox-dns-over-https

**Comment by 0xERR0R** [comment]
DoH is more recent standard, but it is not supported everywhere. For example Android can only DoT (DNS-over-TLS) out of the box. This should also work with traefik, but you must use port 853. If your clients can DoH, just use it.

**Reply by PowershellScripter** [reply]
Great information to know! Thank You. I really only care more so about being able to block ads with blocky and setup as a proxy for my desktop computer at home to connect using the domain behind my traefik proxy.

> DoH is more recent standard, but it is not supported everywhere. For example Android can only DoT (DNS-over-TLS) out of the box. This should also work with traefik, but you must use port 853. If your clients can DoH, just use it.

**Reply by PowershellScripter** [reply]
Has anyone started working on a web ui for blocky? Seems like itd be a fun project.

**EDIT** I mean a web ui for controlling blocky. I know grafana is used for the stats part.

**Reply by 0xERR0R** [reply]
If you want to block ads only at home, you don't need doh, simple dns on port 53 would be enough. Just set blockys ip as dns in your router (dhcp).

Regarding the ui: there are 2 grafana dashboards to see stats and query for dns questions, but it is also possible to disable and enable the blocking state

**Reply by PowershellScripter** [reply]
> If you want to block ads only at home, you don't need doh, simple dns on port 53 would be enough. Just set blockys ip as dns in your router (dhcp).
> 
> Regarding the ui: there are 2 grafana dashboards to see stats and query for dns questions, but it is also possible to disable and enable the blocking state

Im actually setting blocky up for my works network as the local ad block dns and gonna create a doh proxy to connect into from home to block ads in my browser
